ocs_ci/ocs/metrics.py

```diff
@@ -321,3 +325,131 @@ def get_missing_metrics(prometheus, metrics, current_platform=None):
+    return retv
+
+
+def parse_cadvisor_metrics_line(mline):
+    """
+    Parse one of the statistics in the Cadvisor metrics data.
+
+    Parameters:
+        mline -- line of raw output from the Cadvisor metrics data
+
+    Returns:
+        A dictionary with the following entries:
+            stat_name -- The name of the metric passed in this line
+            fields -- A dictionary extracted from the x=y style
+                      parameters passed inside the brackets in mline
+            numbers -- A list of the numbers passed at the end of mline
+    """
+    if mline.startswith("#"):
+        return False
+    if len(mline) == 0:
+        return False
+    split1 = mline.split("{")
+    stat_name = split1[0]
+    try:
+        split2 = split1[1].split("}")
+        fields = parse_cadvisor_extras(split2[0])
+        numbers = split2[1].strip().split(" ")
+    except IndexError:
+        fields = {}
+        nofields_str = stat_name.split(" ")
+        stat_name = nofields_str[0]
+        numbers = nofields_str[1:]
+    nnumbs = []
+    try:
+        for numb in numbers:
+            nnumbs.append(float(numb))
+    except ValueError:
+        logging.info("Invalid number in Cadviser metrics field")
+        return False
+    return {"stat_name": stat_name, "fields": fields, "numbers": nnumbs}
+
+
+def collect_cadvisor_metrics():
+    """
+    Collect a set of statistics from proxy/metrics/cadvisor.
+
+    Returns: a list of dictionary values represent a metric collected.
+             The format of these dictionary values is defined in
+             parse_metrics_line.
+    """
+    ocs = OCP(kind="pod", namespace="openshift-storage")
+    ocs_pods = ocs.get()
+    nlist = []
+    for x in ocs_pods["items"]:
+        pod = x["spec"]["nodeName"]
+        nlist.append(pod)
+    worker_nodes = get_nodes(node_type="worker")
+    stat_list = []
+    for wnode in worker_nodes:
+        result = subprocess.run(
+            [
+                "oc",
+                "get",
+                "--raw",
+                f"/api/v1/nodes/{wnode.name}/proxy/metrics/cadvisor",
+            ],
+            capture_output=True,
+            text=True,
+        )
+        for line in result.stdout.split("\n"):
+            tval = parse_cadvisor_metrics_line(line)
+            if tval:
+                stat_list.append(tval)
+    return stat_list
+
+
+def collect_pod_metrics():
+    """
+    Scan the cadvisor metrics and find all entries for a device on a pod
+
+    Return a dictionary indexed by pod.  The data in each entry is a dict
+            indexed by device.  The device information is indexed by statistic
+            name, and contains a dict of number and operation values.
+    """
+    metrics = collect_cadvisor_metrics()
+    if metrics[0]["stat_name"] == "cadvisor_version_info":
+        logging.info("Found cadvisor metrics")
+    ret_dict = {}
+
+    for minfo in metrics:
+        parms = minfo["fields"]
```

Comment:
Not sure what `parms` stands for. Maybe we can change this to `fields`?

Reply:
Changed to metric_field.